### PR TITLE
Tag service objects with unit tests

### PR DESCRIPTION
**Pull request by fear-the-reaper**
- Added the `TagService, Tag, and IntelOwlClient`
- Implemented proper unit testing and exception handling
- Refactored some of the test code

**This should be merged after the #28 PR**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -29,14 +31,37 @@ jobs:
...
     steps:
     - name: Checkout repository
       uses: actions/checkout@v3
+
+    # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
       with:
         languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
```
Comment (sp35) [inline .github/workflows/codeql.yml:46]: Let's clean up these comments

File: gointelowl/tag.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		return true, nil
+	}
+	return false, nil
+}
+
+//* Pretty printing the tag
+func (tag *Tag) Display() {
+	display := fmt.Sprintf("ID:%d\nLabel:%s\nColor:%s", tag.ID, tag.Label, tag.Color)
+	fmt.Println(display)
+}
```
Comment (sp35) [inline gointelowl/tag.go:167]: Is this method required? let's remove this

File: tests/intelOwl_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/intelowlproject/go-intelowl/gointelowl"
+)
+
+type TestData struct {
+	StatusCode int
+	Data       string
+}
+
+func MakeNewTestServer(testData *TestData) *httptest.Server {
```
Comment (sp35) [inline tests/intelOwl_test.go:16]:
```suggestion
func NewTestServer(testData *TestData) *httptest.Server {
```

Comment (sp35) [reply tests/intelOwl_test.go:16]: Either `Make` or `New` is fine

File: tests/intelOwl_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if testData.StatusCode > 0 {
+			w.WriteHeader(testData.StatusCode)
+		}
+		if len(testData.Data) > 0 {
+			_, err := w.Write([]byte(testData.Data))
+			if err != nil {
+				fmt.Printf("Error: %s", err)
```
Comment (sp35) [inline tests/intelOwl_test.go:24]: Unhandled error, this method should return an error

File: tests/intelOwl_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+			}
+		}
+	}))
+}
+
+func MakeNewTestIntelOwlClient(url string) gointelowl.IntelOwlClient {
```
Comment (sp35) [inline tests/intelOwl_test.go:30]: Either `Make` or `New` is fine

**Review by sp35 — CHANGES_REQUESTED** [review]
Please cherry-pick the actual commits you need in a separate branch!

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (fear-the-reaper) [reply .github/workflows/codeql.yml:46]: Alright!

File: gointelowl/tag.go (hunk 2 above)
Comment (fear-the-reaper) [reply gointelowl/tag.go:167]: I added them for pretty-printing the tags but sure will remove it.

File: tests/intelOwl_test.go (hunk 4 above)
Comment (fear-the-reaper) [reply tests/intelOwl_test.go:24]: will do!

**Pull request closed without merging**